Assets/Scripts/Actions/ActionQueue.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using System.Linq;
+
+namespace DigThemGraves
+{
+    public class ActionQueue : Queue
+    {
+        private LinkedList<ActionModel> actionList = new LinkedList<ActionModel>();
+        private IAction currentAction = null;
+
+        [SerializeField]
+        private GameObject actionMenu;
+
+        [SerializeField]
+        private GameObject actionIcon;
+
+        void Update()
+        {
+            UpdateActions();
+        }
+
+        public bool IsEmpty()
+        {
+            return actionList.Count == 0;
+        }
+
+        private void UpdateActions()
+        {
+            if (currentAction == null)
+            {
+                if (!IsEmpty())
+                {
+                    currentAction = actionList.First.Value.Action;
+                    currentAction.Execute();
+                }
+            }
+            else if (currentAction.IsFinished)
+            {
+                RemoveAction(currentAction);
+                currentAction = null;
```

Comment:
Zostawiłem tak mimo ostrzeżeń, by nie przypisywać do null. Inaczej pojawiają się problemy z wykonywaniem akcji. Poza tym, wydaje mi się, że w tym przypadku pasuje, gdyż jak nie ma akcji, to powinno być puste (null).